src/webhook/webhook-event.handler.ts

```diff
@@ -0,0 +1,89 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable } from '@nestjs/common';
+import { Interval } from '@nestjs/schedule';
+import { Recipe, User, WebhookEventStatus } from '@prisma/client';
+import { lastValueFrom } from 'rxjs';
+import { PrismaService } from '../prisma/prisma.service';
+import { WebhookName } from './enums/webhookName.enum';
+import { WebhookService } from './webhook.service';
+
+@Injectable()
+export class WebhookEventHandler {
+  constructor(
+    private readonly prismaService: PrismaService,
+    private readonly httpService: HttpService,
+    private readonly webhookService: WebhookService,
+  ) {}
+  async createWebhookEvent(
+    webhookName: WebhookName,
+    payload: Recipe | User,
+  ): Promise<void> {
+    const webhook = await this.webhookService.fetchOneWebhookByName(
+      webhookName,
+    );
+
+    await this.prismaService.webhookEvent.create({
+      data: {
+        webhook: {
+          connect: {
+            id: webhook.id,
+          },
+        },
+        payload,
+        retriesLeft: webhook.retriesAmount,
+      },
+    });
+  }
+
+  @Interval(5000)
```

Comment:
Although this would work. I would not do it that way.
I would create a separate module that uses `@Cron()` decorator. 
This module would be responsible for collecting the events stored in the database and resending the webhooks. Please consider that.

Reply:
Also i do not see tests covering the logic of triggerWebhookEvents()